addons/web_tour/static/src/js/tip.js

```diff
@@ -316,8 +316,29 @@ var Tip = Widget.extend({
                 of: this.$anchor,
                 collision: "none",
                 using: props => {
-                    this.el.style.setProperty('top', `${props.top}px`, 'important');
-                    this.el.style.setProperty('left', `${props.left}px`, 'important');
+                    const {top} = props;
+                    let {left} = props;
+                    const hasInnerText = this.$anchor.contents().first().text().trim().length;
+                    if (hasInnerText) {
+                        const textClone = this.$anchor.clone();
+                        const textRulerContainer = textClone.wrap('<div class="o_center_on_text">').parent();
+                        textClone.wrapInner('<span>');
+                        textRulerContainer.appendTo(this.$anchor);
+                        const [textRuler] = this.$anchor.find('span');
+
+                        const textContainerWidth = this.$anchor.innerWidth();
+                        const textWidth = textRuler.getBoundingClientRect().width;
+                        const alignment = this.$anchor.css('text-align');
```

Comment:
Do you actually need to clone and append a new element? Can't you just wrap the real content in a span (with a specific class name so the mutation observer can ignore it like you did with o_center_on_text), I am not sure to understand the point with the clone + absolute positioning